### PR TITLE
[dtc] Run ldconfig after install and uninstall ; fixes JB#45185

### PR DESCRIPTION
**Pull request by uaoh**
Signed-off-by: Aapi Hämäläinen <aapi.hamalainen@jollamobile.com>

### CODE REVIEW

File: rpm/dtc.spec [hunk 1]
```diff
@@ -10,6 +10,7 @@ License:    GPLv2
 
 BuildRequires: flex
 BuildRequires: bison
+Requires:      ldconfig
```
Comment (saukko) [inline rpm/dtc.spec]: This should be: 
`Requires(post): /sbin/ldconfig
Requires(postun): /sbin/ldconfig
`

**Comment by saukko** [comment]
Also it seems that the license should be GPLv2+, because the license headers seems to be mentioning GPLv2 or (at your option) any later version.

File: rpm/dtc.spec (hunk 1 above)
Comment (uaoh) [reply rpm/dtc.spec]: Hmm, glibc provides the cap 'ldconfig', but i guess the '/sbin/ldconfig' works as well -- https://build.merproject.org/package/binary/mer:core/glibc?arch=armv8el&filename=glibc-2.25%2Bgit4-1.14.1.armv7hl.rpm&repository=latest_armv7hl

**Comment by uaoh** [comment]
Updated.

**Comment by uaoh** [comment]
Adapted Fedora packaging;
- https://src.fedoraproject.org/rpms/dtc/blob/master/f/dtc.spec ;
- test build in `home:ahamalainen:branches:pj:hw:qualcomm:f5121` ;
- python bindings disabled with `%if 0%{?dtc_build_python_bindings:1}` ;
- produces
  - `libfdt`,
  - `libfdt-devel`,
  - `libfdt-static`,
  - `dtc`, and optionally
  - `python2-libfdt`.

File: rpm/dtc.spec [hunk 2]
```diff
@@ -1,45 +1,107 @@
...
 
-BuildRequires: flex
-BuildRequires: bison
+BuildRequires:     gcc
+BuildRequires:     make
+BuildRequires:     flex
+BuildRequires:     bison
+BuildRequires:     swig
+
+%if 0%{?dtc_build_python_bindings:1}
```
Comment (saukko) [inline rpm/dtc.spec]: Lets not add python package if we don't need to, less stuff to maintain.

Comment (uaoh) [reply rpm/dtc.spec]: Dropped `python2-libfdt`.

File: rpm/dtc.spec [hunk 3]
```diff
@@ -1,45 +1,82 @@
...
-%{_bindir}/dtdiff
-%{_bindir}/fdtdump
-%{_bindir}/fdtget
-%{_bindir}/fdtoverlay
-%{_bindir}/fdtput
-%{_includedir}/fdt.h
-%{_includedir}/libfdt.h
-%{_includedir}/libfdt_env.h
-%{_libdir}/libfdt-1.4.6.so
+%attr(0644, -, -) %{_defaultlicensedir}/%{name}/GPL
```
Comment (saukko) [inline rpm/dtc.spec]: this should be done with %license macro

File: rpm/dtc.spec [hunk 4]
```diff
@@ -1,45 +1,82 @@
...
+%attr(0644, -, -) %{_defaultdocdir}/%{name}/manual.txt
+%{_bindir}/*
+
+%files -n libfdt
+%attr(0644, -, -) %{_defaultlicensedir}/%{name}/GPL
```
Comment (saukko) [inline rpm/dtc.spec]: As well this, however if the main package depends on this package it is enough to have this license only.

**Comment by uaoh** [comment]
`dtc` doesn't have a runtime dep on `libfdt`, so both deploying the license seems to be correct; moved back to using macros instead of %attr.

**Comment by uaoh** [comment]
Yeah, no `libfdt` spec for `dtc`:
```
..../dtc/binaries$ ldd usr/bin/dtc 
	linux-gate.so.1 (0xf7fad000)
	libc.so.6 => /lib/i386-linux-gnu/libc.so.6 (0xf7d9c000)
	/lib/ld-linux.so.2 (0xf7faf000)
```

**Comment by uaoh** [comment]
Dropped `libfdt*` pacs, so only `dtc` left, with
```
usr/bin/dtc
usr/bin/dtdiff
usr/share/licenses/dtc-1.4.6+jb.45185.20190322191951.1.g51580ab/GPL
usr/share/doc/dtc-1.4.6+jb.45185.20190322191951.1.g51580ab/manual.txt
```
I considered not installing `dtdiff`, but it just might be used by the build process.

File: rpm/dtc.spec [hunk 5]
```diff
@@ -1,45 +1,42 @@
...
-%setup -q
+%autosetup -p1
 
 %build
 cd dtc
-# We only need the compiler, no python necessary so skip it.
-make NO_PYTHON=1 HOME=$RPM_BUILD_ROOT/usr
+make %{?_smp_mflags} V=1 CC="gcc $RPM_OPT_FLAGS $RPM_LD_FLAGS" dtc
+# Drop libfdt bits from manual as we do not install it
+sed -ri '
```
Comment (mkosola) [inline rpm/dtc.spec:32]: sed should be build requirement as it is used in spec file?

Comment (uaoh) [reply rpm/dtc.spec:32]: Yes. Added.

**Review by mkosola — APPROVED** [review]

**Pull request merged**